File: Puzzles/MEDIUM/C/DontPanic_EP1.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main()
{
    int nbFloors; // number of floors
    int width; // width of the area
    int nbRounds; // maximum number of rounds
    int exitFloor; // floor on which the exit is found
    int exitPos; // position of the exit on its floor
    int nbTotalClones; // number of generated clones
    int nbAdditionalElevators; // ignore (always zero)
    int nbElevators; // number of elevators
    scanf("%i%i%i%i%i%i%i%i",&nbFloors, &width, &nbRounds, &exitFloor, &exitPos, &nbTotalClones, &nbAdditionalElevators, &nbElevators);
    int elevatorFloor[nbElevators]; // floor on which this elevator is found
    int elevatorPos[nbElevators]; // position of the elevator on its floor
    for (int i = 0; i < nbElevators; i++) {

        scanf("%i%i", &elevatorFloor[i], &elevatorPos[i]);
    }

    // game loop
    while (1) {
        int cloneFloor; // floor of the leading clone
        int clonePos; // position of the leading clone on its floor
        char direction[11]; // direction of the leading clone: LEFT or RIGHT
        scanf("%d%d%s", &cloneFloor, &clonePos, direction);

        if(cloneFloor==exitFloor||cloneFloor==-1)
        {
            if(exitPos<clonePos){
                if(direction[0]=='L'){
                    printf("WAIT\n");
                }
                else{
                    printf("BLOCK\n");
                }
            }
            else{
                if(direction[0]=='R'){
                    printf("WAIT\n");
                }
                else{
                    printf("BLOCK\n");
                }
            }
        }
        else if (cloneFloor<exitFloor){
            int NextEl;
            for(int i=0;i<nbElevators;i++){
                if(cloneFloor==elevatorFloor[i]){
                    NextEl=i;
                }
            }
            if(elevatorPos[NextEl]<clonePos){
                if(direction[0]=='L'){
                    printf("WAIT\n");
                }
                else{
                    printf("BLOCK\n");
                }
            }
            else if(elevatorPos[NextEl]>clonePos){
                if(direction[0]=='R'){
                    printf("WAIT\n");
                }
                else{
                    printf("BLOCK\n");
                }
            }
            else if(elevatorPos[NextEl]==clonePos){
                printf( "WAIT\n");
            }
        }        
    }       
}

File: Puzzles/MEDIUM/PYTHON/DontPanic_EP1.py
elevator_pos, elevator_input, WAIT, BLOCK = {}, [], "WAIT", "BLOCK"
(
    nb_floors,
    width,
    nb_rounds,
    exit_floor,
    exit_pos,
    nb_total_clones,
    nb_additional_elevators,
    nb_elevators,
) = [int(i) for i in input().split()]

for i in range(nb_elevators):
    elevator_input.append([int(j) for j in input().split()])
    elevator_pos[elevator_input[i][0]] = elevator_input[i][1]

while 1:
    clone_floor, clone_pos, direction = input().split()
    clone_floor, clone_pos = int(clone_floor), int(clone_pos)

    if clone_floor == exit_floor or clone_floor == -1:
        command = (WAIT, BLOCK)[
            (clone_pos > exit_pos and direction[:1] == "R")
            or (clone_pos < exit_pos and direction[:1] == "L")
        ]
    else:
        command = (WAIT, BLOCK)[
            (clone_pos > elevator_pos[clone_floor] and direction[:1] == "R")
            or (clone_pos < elevator_pos[clone_floor] and direction[:1] == "L")
        ]
        
    print(command)
